### ISSUE TITLE
Chef 12.4.0 and 12.4.1 - Chef::Exceptions::UnsupportedAction: Setting members directly is not supported by

### ISSUE THREAD

**Issue by dzabel** [chef/chef#3780]
```
Chef::Exceptions::UnsupportedAction: group[root] (/var/chef/cache/cookbooks/users/providers/manage.rb line 160) had an error: Chef::Exceptions::UnsupportedAction: Setting members directly is not supported by #<Chef::Provider::Group::Usermod:0x1989ca8>
```
this occurs in conjunction with users cookbook - only this is done:

```
users_manage 'root' do
    action [:remove, :create]
end
```

going back to chef 12.4.0-1 fixes this.

**Comment by ranjib** [comment]
@dzabel can you share the stack trace?

**Comment by dzabel** [comment]
output chef-client run:

```
      ================================================================================
      Error executing action `create` on resource 'group[root]'
      ================================================================================

      Chef::Exceptions::UnsupportedAction
      -----------------------------------
      Setting members directly is not supported by #<Chef::Provider::Group::Usermod:0x998e514>

      Resource Declaration:
      ---------------------
      # In /var/chef/cache/cookbooks/users/providers/manage.rb

      160:   group new_resource.group_name do
      161:     if new_resource.group_id
      162:       gid new_resource.group_id
      163:     end
      164:     members security_group

      Compiled Resource:
      ------------------
      # Declared in /var/chef/cache/cookbooks/users/providers/manage.rb:160:in `block in class_from_file'

      group("root") do
        action [:create]
        retries 0
        retry_delay 2
        default_guard_interpreter :default
        group_name "root"
        gid 0
        members ["root"]
        declared_type :group
        cookbook_name "infrastructure"
      end


    ================================================================================
    Error executing action `create` on resource 'users_manage[root]'
    ================================================================================

    Chef::Exceptions::UnsupportedAction
    -----------------------------------
    group[root] (/var/chef/cache/cookbooks/users/providers/manage.rb line 160) had an error: Chef::Exceptions::UnsupportedAction: Setting members directly is not supported by #<Chef::Provider::Group::Usermod:0x998e514>

    Resource Declaration:
    ---------------------
    # In /var/chef/cache/cookbooks/coremedia-infrastructure/recipes/root.rb

     21:   users_manage 'root' do
     22:     action [:remove, :create]
     23:     not_if { amazon? or windows? }
     24:   end
     25: end

    Compiled Resource:
    ------------------
    # Declared in /var/chef/cache/cookbooks/coremedia-infrastructure/recipes/root.rb:21:in `from_file'

    users_manage("root") do
      action [:remove, :create]
      retries 0
      retry_delay 2
      default_guard_interpreter :default
      declared_type :users_manage
      cookbook_name "infrastructure"
      recipe_name "root"
      data_bag "users"
      search_group "root"
      cookbook "users"
      group_name "root"
      not_if { #code block }
    end
```

output from **/var/chef/cache/chef-stacktrace.out**

```
Generated at 2015-09-09 09:33:55 +0200
Chef::Exceptions::UnsupportedAction: users_manage[root] (infrastructure::root line 21) had an error: Chef::Exceptions::UnsupportedAction: group[root] (/var/chef/cache/cookbooks/users/providers/manage.rb line 160) had an error: Chef::Exceptions::UnsupportedAction: Setting members directly is not supported by #<Chef::Provider::Group::Usermod:0x998e514>
/opt/chef/embedded/apps/chef/lib/chef/provider/group/usermod.rb:65:in `set_members'
/opt/chef/embedded/apps/chef/lib/chef/provider/group/groupadd.rb:94:in `modify_group_members'
/opt/chef/embedded/apps/chef/lib/chef/provider/group/groupadd.rb:59:in `manage_group'
/opt/chef/embedded/apps/chef/lib/chef/provider/group.rb:135:in `block in action_create'
/opt/chef/embedded/apps/chef/lib/chef/mixin/why_run.rb:52:in `call'
/opt/chef/embedded/apps/chef/lib/chef/mixin/why_run.rb:52:in `add_action'
/opt/chef/embedded/apps/chef/lib/chef/provider.rb:175:in `converge_by'
/opt/chef/embedded/apps/chef/lib/chef/provider/group.rb:134:in `action_create'
/opt/chef/embedded/apps/chef/lib/chef/provider.rb:140:in `run_action'
/opt/chef/embedded/apps/chef/lib/chef/resource.rb:584:in `run_action'
/opt/chef/embedded/apps/chef/lib/chef/runner.rb:49:in `run_action'
/opt/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `block (2 levels) in converge'
/opt/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `each'
/opt/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `block in converge'
/opt/chef/embedded/apps/chef/lib/chef/resource_collection/resource_list.rb:83:in `block in execute_each_resource'
/opt/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:116:in `call'
/opt/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:116:in `call_iterator_block'
/opt/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:85:in `step'
/opt/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:104:in `iterate'
/opt/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:55:in `each_with_index'
/opt/chef/embedded/apps/chef/lib/chef/resource_collection/resource_list.rb:81:in `execute_each_resource'
/opt/chef/embedded/apps/chef/lib/chef/runner.rb:80:in `converge'
/opt/chef/embedded/apps/chef/lib/chef/provider/lwrp_base.rb:62:in `recipe_eval_with_update_check'
/opt/chef/embedded/apps/chef/lib/chef/provider/lwrp_base.rb:46:in `block in action'
/opt/chef/embedded/apps/chef/lib/chef/provider.rb:140:in `run_action'
/opt/chef/embedded/apps/chef/lib/chef/resource.rb:584:in `run_action'
/opt/chef/embedded/apps/chef/lib/chef/runner.rb:49:in `run_action'
/opt/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `block (2 levels) in converge'
/opt/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `each'
/opt/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `block in converge'
/opt/chef/embedded/apps/chef/lib/chef/resource_collection/resource_list.rb:83:in `block in execute_each_resource'
/opt/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:116:in `call'
/opt/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:116:in `call_iterator_block'
/opt/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:85:in `step'
/opt/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:104:in `iterate'
/opt/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:55:in `each_with_index'
/opt/chef/embedded/apps/chef/lib/chef/resource_collection/resource_list.rb:81:in `execute_each_resource'
/opt/chef/embedded/apps/chef/lib/chef/runner.rb:80:in `converge'
/opt/chef/embedded/apps/chef/lib/chef/client.rb:654:in `block in converge'
/opt/chef/embedded/apps/chef/lib/chef/client.rb:649:in `catch'
/opt/chef/embedded/apps/chef/lib/chef/client.rb:649:in `converge'
/opt/chef/embedded/apps/chef/lib/chef/client.rb:688:in `converge_and_save'
/opt/chef/embedded/apps/chef/lib/chef/client.rb:269:in `run'
/opt/chef/embedded/apps/chef/lib/chef/application.rb:270:in `block in fork_chef_client'
/opt/chef/embedded/apps/chef/lib/chef/application.rb:258:in `fork'
/opt/chef/embedded/apps/chef/lib/chef/application.rb:258:in `fork_chef_client'
/opt/chef/embedded/apps/chef/lib/chef/application.rb:224:in `block in run_chef_client'
/opt/chef/embedded/apps/chef/lib/chef/local_mode.rb:39:in `with_server_connectivity'
/opt/chef/embedded/apps/chef/lib/chef/application.rb:212:in `run_chef_client'
/opt/chef/embedded/apps/chef/lib/chef/application/client.rb:402:in `block in interval_run_chef_client'
/opt/chef/embedded/apps/chef/lib/chef/application/client.rb:392:in `loop'
/opt/chef/embedded/apps/chef/lib/chef/application/client.rb:392:in `interval_run_chef_client'
/opt/chef/embedded/apps/chef/lib/chef/application/client.rb:382:in `run_application'
/opt/chef/embedded/apps/chef/lib/chef/application.rb:60:in `run'
/opt/chef/embedded/apps/chef/bin/chef-client:26:in `<top (required)>'
/usr/bin/chef-client:55:in `load'
/usr/bin/chef-client:55:in `<main>'
```

**Comment by thommay** [comment]
@dzabel can you confirm which platform? The results of `ohai platform` and `ohai platform_family` will be  useful.

**Comment by dzabel** [comment]
SunOS 5.11

solaris_x86:
```
  "platform_version": "5.11",
  "platform_build": "11.1",
  "platform": "solaris2",
  "platform_family": "solaris2",
```

solaris sparc:
```
  "platform_version": "5.11",
  "platform_build": "11.1",
  "platform": "solaris2",
  "platform_family": "solaris2",
```

Both are affected.

**Comment by tas50** [comment]
This is occurring with Chef 12.12 on OpenSUSE 13.2 as well.

```
    * group[cool_group] action create

           ================================================================================
           Error executing action `create` on resource 'group[cool_group]'
           ================================================================================

           Chef::Exceptions::UnsupportedAction
           -----------------------------------
           Setting members directly is not supported by #<Chef::Provider::Group::Usermod:0x00000003c3f908>

           Cookbook Trace:
           ---------------
           /tmp/kitchen/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:41:in `run_action'

           Resource Declaration:
           ---------------------
           # In /tmp/kitchen/cache/cookbooks/test/recipes/helloworld_example.rb

             7: group 'cool_group' do
             8:   members 'chefed'
             9:   action :create
            10: end
            11:
```

**Comment by tas50** [comment]
@dzabel The fix is in and this will ship with Chef 12.14
